dotnet/src/SemanticKernel/Planning/ConditionException.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Planning;
+public class ConditionException : Exception<ConditionException.ErrorCodes>
+{
+    /// <summary>
+    /// Error codes for <see cref="ConditionException"/>.
+    /// </summary>
+    public enum ErrorCodes
+    {
+        /// <summary>
+        /// Unknown error.
+        /// </summary>
+        UnknownError = -1,
```

Comment:
In C# default enum value is zero, so I would mark UnknownError as `0`, just in case, so `default(ErrorCodes)` will return `UnknownError` instead of `InvalidConditionFormat`.

Reply:
Agree with you but the existing convention in SK all the Exception errorcodes.UnknownError are -1.